ts/Extensions/Annotations/Annotations.ts

```diff
@@ -179,6 +179,8 @@ declare global {
             markerStart?: string;
             point?: (string|AnnotationMockPointOptionsObject);
             points?: Array<(string|AnnotationMockPointOptionsObject)>;
+            xAxis: number;
```

Comment:
It's index. I didn't change the name because it was named like this previously. We can consider changing it, but there are so many places when an axis' index is called this way that I think it should be a part of potential Stock Tools refactoring in the future.

Reply:
It's the same as with option `series.xAxis` or `series.yAxis`. It's an index, even when the name does not suggest it.

However, do we have `shapeOptions.xAxis` or `shapes[index].xAxis`? Only `shapes[index].point` or `points` have this option, according to the API: https://api.highcharts.com/highstock/annotations.shapes.point.xAxis (commented with `(1)` the same places)

And the format is `number | string | null`, could you double-check this?

Reply:
> And the format is number | string | null, could you double-check this?

When `series.xAxis` or `series.yAxis` is a string, it refers to the `id` of the axis. For consistency, I think we should recreate that feature for the Annotations too. 

Also, I think we can simplify a bit and disallow `null`. I think 0 has the same effect, right?

Reply:
Nice catch, @pawelfus 
I corrected TS definitions and now I'm taking axis index from shapes[index] point. Also, the API description should be corrected or even removed (the user doesn't need to define https://api.highcharts.com/highstock/annotations.shapes.point.xAxis because the click finds a proper axis automatically.

Reply:
Nice catch, @pawelfus
I corrected TS definitions and now I'm taking axis index from shapes[index] point. Also, the API description should be corrected or even removed (the user doesn't need to define https://api.highcharts.com/highstock/annotations.shapes.point.xAxis because the click finds a proper axis automatically.

Reply:
I think this option can be removed from API anyway because now the user doesn't need to define on which axis he is adding an annotation - new functionality finds the proper axis automatically.

When it comes to TS definitions, I think the Annotation's MockPoint's `xAxis` or `yAxis` props are only `number | AxisType` type now. I removed `string` and `null` from TS definitions.

Are you OK with these changes? https://github.com/highcharts/highcharts/pull/15209/commits/fd88d5ef63d43b515446d12f904475c0faed7b23